htdocs/core/class/html.form.class.php

```diff
@@ -6872,19 +6872,26 @@ public static function multiselectarray($htmlname, $array, $selected = array(),
 				$array = array_combine($array, $array);
 			}
 
-			if (!empty($array)) {
-				foreach ($array as $key => $value) {
+			if (! empty($array))
+			{
+				foreach ($array as $key => $value)
+				{
+					if (count(explode("|", $value)) == 2){
+						$info = explode("|", $value);
+						$out.= '<option value="'.$key.'" parent="'.$info[1].'"';
+						$value = $info[0];
+					} else {
+						$out .= '<option value="' . $key . '"';
+					}
+                    if (is_array($selected) && ! empty($selected) && in_array((string) $key, $selected) && ((string) $key != ''))
+					{
+						$out.= ' selected';
+					}
+					$out.= '>';
 					$newval = ($translate ? $langs->trans($value) : $value);
 					$newval = ($key_in_label ? $key.' - '.$newval : $newval);
-
-					$out .= '<option value="'.$key.'"';
-					if (is_array($selected) && !empty($selected) && in_array((string) $key, $selected) && ((string) $key != '')) {
-						$out .= ' selected';
-					}
-					$out .= ' data-html="'.dol_escape_htmltag($newval).'"';
```

Comment:
The data-html tag seems to have disappeared. This create regression making html beautified select list not working anymore.